scalding-macros/src/main/scala/com/twitter/scalding/macros/impl/ordered_serialization/providers/StringOrderedBuf.scala

```diff
@@ -0,0 +1,122 @@
+/*
+ Copyright 2014 Twitter, Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.twitter.scalding.macros.impl.ordered_serialization.providers
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+import com.twitter.scalding._
+import com.twitter.scalding.macros.impl.ordered_serialization.{ CompileTimeLengthTypes, ProductLike, TreeOrderedBuf }
+import CompileTimeLengthTypes._
+import java.nio.ByteBuffer
+import com.twitter.scalding.serialization.OrderedSerialization
+
+object StringOrderedBuf {
```

Comment:
Scalacheck generates strings by taking random characters and putting them in arrays. It could be more robust, but it is pretty good. The downside is it is biased pretty strongly toward non-ascii strings by default. It would be nice to actually have a large corpus of tweets to use as strings since they should cover a variety of languages.